apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -179,6 +179,8 @@ def read(
                     lo_hi = util.slice_to_range(ids)
                     if lo_hi is not None:
                         sr.set_dim_ranges(SOMA_ROWID, [lo_hi])
+                else:
```

Comment:
As noted in #485, these would be better thrown as a TypeError, as this is a classic use case for that exception.  Per the Python docs for [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError):
> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

Reply:
Gah. I keep changing those back to `TypeError`. I suspect they're getting reset every time I `git merge main`. Sorry about that.

Reply:
force-merge is a big, big hammer